Tasks/DeployAzureResourceGroup/Environment.ts

```diff
@@ -204,5 +205,207 @@ export class RegisterEnvironment {
             this.InstantiateEnvironment();
         });
     }
-   
+    
+    private AddInboundNetworkSecurityRule(retryCnt: number, securityGrpName, networkClient, ruleName, rulePriority, winrmHttpsPort){
+           try{
+               console.log("Adding inbound network security rule config %s with priority %s for port %s under security group %s", ruleName, rulePriority, winrmHttpsPort, securityGrpName);
+               var securityRuleParameters = {direction: "Inbound", access: "Allow", sourceAddressPrefix: "*", sourcePortRange: "*", destinationAddressPrefix: "*", destinationPortRange: winrmHttpsPort, protocol: "*", priority: rulePriority};
+                           
+               networkClient.securityRules.createOrUpdate(this.resourceGroupName, securityGrpName, ruleName, securityRuleParameters, (error, result, request, response)=>{
+                   if(error){
+                       console.log("Error in adding network security rule %s", util.inspect(error, {depth: null}));
+                       throw "FailedToAddRuleToNetworkSecurityGroup";
+                    }    
+                    console.log("Added inbound network security rule config %s with priority %s for port %s under security group %s with result: %s", ruleName, rulePriority, winrmHttpsPort, securityGrpName, util.inspect(result, {depth: null}));              
+                });
+           }
+            catch(exception){
+                console.log("Failed to add inbound network security rule config %s with priority %s for port %s under security group %s: %s", ruleName, rulePriority, winrmHttpsPort, securityGrpName, exception.message);
+                rulePriority = rulePriority + 50;
+                console.log("Getting network security group %s in resource group %s", securityGrpName, this.resourceGroupName);
+                networkClient.networkSecurityGroups.list(this.resourceGroupName, (error, result, request, response)=>{
+                    if(error){
+                        console.log("Error in getting the list of network Security Groups for the resource-group %s", this.resourceGroupName);
+                        throw "FetchingOfNetworkSecurityGroupFailed";
+                    }
+                    if(result.length > 0){
+                        console.log("Got network security group %s in resource group %s", securityGrpName, this.resourceGroupName);
+                        if(retryCnt>0){
+                            retryCnt = retryCnt - 1;
+                            this.AddInboundNetworkSecurityRule(retryCnt, securityGrpName, networkClient, ruleName, rulePriority, winrmHttpsPort);
```

Comment:
You can directly give this.AddInboundNetworkSecurityRule(retryCnt -1 , securityGrpName, networkClient, ruleName, rulePriority, winrmHttpsPort);